### PR TITLE
Introduce a Team Monthly plan

### PR DESCRIPTION
**Pull request by stephencelis**
mar-04-2018 12-25-52

This allows folks to sign up for a team plan on a monthly billing basis! Still need some work on the account page to allow for upgrades/downgrades/crossgrades, but that'll come soon!

### CODE REVIEW

File: Sources/PointFree/Pricing.swift [hunk 1]
```diff
@@ -10,11 +10,11 @@ import Prelude
 import Styleguide
 import Tuple
 
-public enum Pricing: Codable, DerivePartialIsos {
-  case individual(Billing)
-  case team(Int)
+public struct Pricing {
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: Don't mind me just using algebra with my types.

File: Sources/PointFree/Pricing.swift [hunk 2]
```diff
@@ -25,90 +25,64 @@ public enum Pricing: Codable, DerivePartialIsos {
...
-    switch self {
-    case .team:
-      return true
-    case .individual:
-      return false
+    return self.lane == .team
+  }
+}
+
+extension Pricing: Codable {
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: Being a struct means we could auto-derive here, but we do some fancy things on the subscription page with shown/not-shown fields, so this is a bit safer.

File: Sources/PointFree/Pricing.swift [hunk 3]
```diff
@@ -165,12 +139,13 @@ let pricingOptionsView = View<(Database.User?, Pricing, Bool)> { currentUser, pr
...
                 id(Stripe.html.formId),
                 method(.post),
-                onsubmit(javascript: "event.preventDefault();")
+                onsubmit(javascript: "event.preventDefault()")
               ],
               pricingTabsView.view(pricing)
-                + individualPricingRowView.view(pricing)
-                + teamPricingRowView.view(pricing)
-                + pricingFooterView.view((currentUser, expand))
+                <> [div([`class`([Class.margin([.mobile: [.bottom: 3]])])], [])]
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: Just letting a semigroup be a semigroup.

File: Sources/PointFree/Pricing.swift [hunk 4]
```diff
@@ -274,7 +249,7 @@ private let whoAreYou = View<Prelude.Unit> { _ in
         "We’re ",
         a([href("http://www.fewbutripe.com"), style(faqLinkStyles)], ["Brandon Williams"]),
         " and ",
-        a([href("http://www.stephencelis.com"), style(faqLinkStyles)], ["Stephen Celis"]),
+        a([href("https://www.stephencelis.com"), style(faqLinkStyles)], ["Stephen Celis"]),
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: FYI.

File: Sources/PointFree/Pricing.swift [hunk 5]
```diff
@@ -358,91 +349,86 @@ private func isChecked(_ billing: Pricing.Billing, _ pricing: Pricing) -> Bool {
...
+          id(radioId(for: billing)),
+          name("pricing[billing]"),
           type(.radio),
-          value($0.billing.rawValue),
+          value(billing.rawValue),
           ]),
         gridColumn(sizes: [.mobile: 12], [], [
-          h2([`class`([Class.pf.type.title2, Class.type.light, Class.pf.colors.fg.gray650])], [.text(encode(pricingText(for: $0.billing)))]),
+          h2([`class`([Class.pf.type.responsiveTitle2, Class.type.light, Class.pf.colors.fg.gray650])], [
+            span([`class`([selectors.content.0])], [
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: Scoping these using CSS still to avoid writing more JavaScript...

File: Sources/PointFree/Pricing.swift [hunk 6]
```diff
@@ -536,28 +540,28 @@ private let selectors = (
     CssSelector.id("tab1")
   ),
   content: (
-    CssSelector.id("content0"),
-    CssSelector.id("content1")
+    CssSelector.class("content0"),
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: Changed these to use `class` selectors so we can hide/show things more granularly.

File: Sources/PointFree/Pricing.swift [hunk 7]
```diff
@@ -536,28 +540,28 @@ private let selectors = (
...
+    CssSelector.class("content1")
   )
 )
 
 private func tabStyles(
-  idSelectors: [(input: CssSelector, content: CssSelector)],
-  showTabStyles: Stylesheet = display(.flex)
+  selectors: [(input: CssSelector, content: CssSelector)],
+  showTabStyles: Stylesheet = display(.inherit)
```
Comment (stephencelis) [inline Sources/PointFree/Pricing.swift]: This is kind of a hack, but works! Just need to make sure the parent node uses the proper display.

File: Sources/PointFree/Pricing.swift (hunk 1 above)
Comment (mbrandonw) [reply Sources/PointFree/Pricing.swift]: lol

File: Sources/PointFree/Pricing.swift [hunk 8]
```diff
@@ -10,11 +10,11 @@ import Prelude
...
-  case individual(Billing)
-  case team(Int)
+public struct Pricing {
+  let billing: Billing
+  let quantity: Int
 
-  public static let `default` = individual(.monthly)
+  public static let `default` = Pricing(billing: .monthly, quantity: 1)
 
   public static let teamYearlyBase = 160
```
Comment (mbrandonw) [inline Sources/PointFree/Pricing.swift]: this caught my eye, doesn't actually seem to be used anymore...

File: Sources/PointFree/Pricing.swift (hunk 6 above)
Comment (mbrandonw) [reply Sources/PointFree/Pricing.swift]: ah nice, that works! cool that we have the screenshot tests that execute JS to switch tabs!

File: Tests/PointFreeTests/PricingTests.swift [hunk 9]
```diff
@@ -35,9 +35,9 @@ class PricingTests: TestCase {
         
         webView.evaluateJavaScript(
           """
-          document.getElementById("tab0").checked = false;
-          document.getElementById("tab1").checked = true;
-          var quantity = document.getElementsByName("pricing[team]")[0];
+          document.getElementById('tab0').checked = false;
+          document.getElementById('tab1').checked = true;
+          var quantity = document.getElementsByName('pricing[quantity]')[0];
```
Comment (mbrandonw) [inline Tests/PointFreeTests/PricingTests.swift]: lol I heard that single quotes are more performant than double quotes......

jkjk

Comment (stephencelis) [reply Tests/PointFreeTests/PricingTests.swift]: Lolol we use single quotes elsewhere so...consistency dragging me. Makes it at least nice to do non-multiline JS where we'd have to escape all `"`s.

**Pull request merged**